src/main/java/edu/byu/cs/autograder/git/GitHelper.java

```diff
@@ -71,12 +99,18 @@ private void fetchRepo(File intoDirectory) throws GradingException {
     }
 
     // Early decisions
-    private CommitVerificationResult skipCommitVerification(File stageRepo) throws GradingException {
-        LOGGER.debug("Skipping commit verification");
+    private CommitVerificationResult skipCommitVerification(boolean verified, File stageRepo) throws GradingException {
         String headHash = getHeadHash(stageRepo);
+        return skipCommitVerification(verified, headHash, null);
+    }
+    private CommitVerificationResult skipCommitVerification(boolean verified, @NonNull String headHash, String failureMessage) {
+        if (headHash == null) {
+            throw new IllegalArgumentException("Head hash cannot be null");
+        }
```

Comment:
Same question about headHash being null. I'd rather not have a student end up with an error message that just says "Head hash cannot be null" and wondering what happened

Reply:
If I don't but the check here, the SQL will throw the for me when it attempts to save in the database. This check just allows the computer to print out a helpful stack trace with a named error to help a developer track down the actual error.